### PR TITLE
Bump msrv and ensure tests work with msrv

### PR DESCRIPTION
**Pull request by nicklan**
## What changes are proposed in this pull request?

`cargo msrv` doesn't seem to be able to check tests, so we add a new workflow to make sure `cargo test` at least works with our `rust-version`.

Incidentally, we need to bump to 1.81.0 for a test.

## How was this change tested?

- PR created without bumped version
- See [job fails](https://github.com/delta-io/delta-kernel-rs/actions/runs/12286349522/job/34286300782?pr=590) before version bump
- bump version
- See [test succeeds](https://github.com/delta-io/delta-kernel-rs/actions/runs/12286434173/job/34286554172?pr=590)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -41,6 +41,30 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install minimal stable and cargo msrv
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: default
+          toolchain: stable
+          override: true
+      - uses: Swatinem/rust-cache@v2
```
Comment (zachschuermann) [inline .github/workflows/build.yml:54]: FYI @roeap will need to make his changes over here too :)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -41,6 +41,30 @@ jobs:
           cargo msrv --path derive-macros/ verify --all-features
           cargo msrv --path ffi/ verify --all-features
           cargo msrv --path ffi-proc-macros/ verify --all-features
+  msrv-run-tests:
```
Comment (zachschuermann) [inline .github/workflows/build.yml:44]: ideally i would say to just put this in `msrv` job above? but i guess we can't do that due to this using specific toolchain and not necessarily wanting that for the job above?

**Review by zachschuermann — APPROVED** [review]
left a question but LGTM

**Comment by zachschuermann** [comment]
also heads up looks like new job failed to get the version for toolchain:
```
error: error: invalid value '$(cargo msrv show --path kernel/ --output-format minimal)' for '<toolchain>...': invalid toolchain name: '$(cargo msrv show --path kernel/ --output-format minimal)'
```

File: .github/workflows/build.yml (hunk 2 above)
Comment (nicklan) [reply .github/workflows/build.yml:44]: yeah, i also want to parallelize this

**Comment by nicklan** [comment]
> also heads up looks like new job failed to get the version for toolchain:
> 
> ```
> error: error: invalid value '$(cargo msrv show --path kernel/ --output-format minimal)' for '<toolchain>...': invalid toolchain name: '$(cargo msrv show --path kernel/ --output-format minimal)'
> ```

Yeah. I'll request when it's ready. I expect to fight with actions a bit before I get this working

**Comment by zachschuermann** [comment]
> > also heads up looks like new job failed to get the version for toolchain:
> > ```
> > error: error: invalid value '$(cargo msrv show --path kernel/ --output-format minimal)' for '<toolchain>...': invalid toolchain name: '$(cargo msrv show --path kernel/ --output-format minimal)'
> > ```
> 
> Yeah. I'll request when it's ready. I expect to fight with actions a bit before I get this working

haha sounds good - sorry i jumped the gun a bit

**Review by zachschuermann — APPROVED** [review]
restamp

**Comment by nicklan** [comment]
closing in favor of #596

**Pull request closed without merging**